complex/cargo/src/lib.rs

```diff
@@ -54,3 +54,31 @@ pub unsafe extern "C" fn toodle_list(toodle: *mut Toodle) -> *mut Arc<ListManage
     let toodle = &*toodle;
     Box::into_raw(Box::new(toodle.list.clone()))
 }
+
+// TODO these interfaces probably belong in separate platform-specific "interface" crates.
+#[cfg(target_os="android")]
+#[allow(non_snake_case)]
+pub mod android {
+    extern crate jni;
+
+    use super::*;
+    use self::jni::JNIEnv;
+    use self::jni::objects::{JClass, JString};
+    use self::jni::sys::{jlong};
+    use ffi_utils::android::log;
+
+    #[no_mangle]
+    pub unsafe extern fn Java_com_mozilla_toodle_RustToodle_newToodle(env: JNIEnv, _: JClass, db_path: JString) -> jlong {
+        log("newToodle!");
```

Comment:
Why not just pass the args into the existing `new_toodle` function to save having to replicate code? I think the fewer places we have to make changes the better.